### PR TITLE
User can only view a part of premium articles

### PR DESCRIPTION
**Pull request by mantanquzzmo**
[PT-Link](https://www.pivotaltracker.com/story/show/170646678)

As a site owner,
In order to get more subscribers,
I would like to restrict all content to paid subscribers only

- Adds condition to show serializer

### CODE REVIEW

File: app/serializers/articles/show_serializer.rb [hunk 1]
```diff
@@ -1,3 +1,19 @@
...
+  include ActionView::Helpers::TextHelper
+
   attributes :id, :title, :body
-end
\ No newline at end of file
+
+  def body
+    if current_user != nil 
+      if current_user.subscriber? || current_user.journalist? || current_user.publisher?
+      object.body
```
Comment (sundqvistjohan) [inline app/serializers/articles/show_serializer.rb:8]: indentation

File: spec/requests/api/v1/articles/show_spec.rb [hunk 2]
```diff
@@ -1,6 +1,14 @@
 RSpec.describe 'GET /api/v1/articles/:id', type: :request do
   let(:headers) { { HTTP_ACCEPT: 'application/json' } }
-  let!(:article) { create(:article)}
+  let!(:article) { create(:article,
+                          title: 'Breaking News',
+                          body: 'Some breaking but also long content.' *  10 )}
```
Comment (EevanR) [inline spec/requests/api/v1/articles/show_spec.rb]: factory can handle title and body?

Comment (mantanquzzmo) [reply spec/requests/api/v1/articles/show_spec.rb]: done

File: app/serializers/articles/show_serializer.rb [hunk 3]
```diff
@@ -1,3 +1,19 @@
...
+      else
+        truncate(object.body, length: 350)
+      end
+    else
+      truncate(object.body, length: 350)
+    end
+  end
```
Comment (sundqvistjohan) [inline app/serializers/articles/show_serializer.rb]: Could be rewritten as 
```
if current_user != nil && current_user.role != nil
  object.body
else
  truncate(object.body, length: 350)
end
```
Which may or mat not be easier to read (it's less explicit who we're allowing the full content). Up to you.

**Review by sundqvistjohan — CHANGES_REQUESTED** [review]

File: app/serializers/articles/show_serializer.rb (hunk 3 above)
Comment (mantanquzzmo) [reply app/serializers/articles/show_serializer.rb]: fixed

File: app/serializers/articles/show_serializer.rb (hunk 1 above)
Comment (mantanquzzmo) [reply app/serializers/articles/show_serializer.rb:8]: fixed

**Comment by raomanasa** [comment]
LGTM

**Review by sundqvistjohan — APPROVED** [review]

File: app/serializers/articles/show_serializer.rb [hunk 4]
```diff
@@ -1,3 +1,15 @@
 class Articles::ShowSerializer < ActiveModel::Serializer
+  include ActionView::Helpers::TextHelper
+
   attributes :id, :title, :body
-end
\ No newline at end of file
+
+  def body
+    if current_user != nil && current_user.role != nil
```
Comment (oliverochman) [inline app/serializers/articles/show_serializer.rb:7]: should a user with the role of `:user` see the whole article?

Comment (mantanquzzmo) [reply app/serializers/articles/show_serializer.rb:7]: We have set default role to nil in previous PR. :user is no more.

Comment (oliverochman) [reply app/serializers/articles/show_serializer.rb:7]: ok cool

**Pull request merged**